src/main/java/io/github/moulberry/notenoughupdates/util/Utils.java

```diff
@@ -252,46 +252,40 @@ public static String shortNumberFormat(double n, int iteration) {
                 : shortNumberFormat(d, iteration + 1));
     }
 
-    public static String trimIgnoreColour(String str) {
-        return trimIgnoreColourStart(trimIgnoreColourEnd(str));
-    }
-
-    public static String trimIgnoreColourStart(String str) {
-        str = str.trim();
-        boolean colourCodeLast = false;
-        StringBuilder colours = new StringBuilder();
-        for (int i = 0; i < str.length(); i++) {
-            char c = str.charAt(i);
-            if (colourCodeLast) {
-                colours.append('\u00a7').append(c);
-                colourCodeLast = false;
+    public static String trimWhitespaceAndFormatCodes(String str) {
+        int startIndex = indexOfFirstNonWhitespaceNonFormatCode(str);
+        int endIndex = lastIndexOfNonWhitespaceNonFormatCode(str);
+        if (startIndex == -1 || endIndex == -1) return "";
+        return str.substring(startIndex, endIndex);
+    }
+
+    private static int indexOfFirstNonWhitespaceNonFormatCode(String str)
+    {
+        int len = str.length();
+        for (int i = 0; i < len; i++) {
+            char ch = str.charAt(i);
+            if (ch <= ' ') {
```

Comment:
You should probably use `Character.isWhitespace(ch)` not `ch <= ' '` as `ch <= ' '` actually catches some none whitespace characters

Reply:
I'm happy to change this, but I intentionally went with `ch <= ' '` since that is equivalent to what String's `trim()` method does that is called by `trimIgnoreColour()` based on the Azul 1.8.0_312 JDK source.

Reply:
I guess it's fine if it's trying to emulate `trim` but the name of `indexOfFirstNonWhitespaceNonFormatCode` is not very evident of that since `trim` only does a very small subsection of space chars and end and start line chars and the name implies other whitespaces if you want to see a method that is similar to what the method name implies you can look at the method of strip in java 11 which does whitespace, not just a subset of spaces.

Reply:
Good point on the naming not being evident of what it was doing. I switched to isWhitespace since that's the use case I would expect to see more.